icu4c/source/common/cmemory.h

```diff
@@ -771,14 +800,29 @@ class MaybeStackVector : protected MemoryPool<T, stackCapacity> {
         return this->create(args...);
     }
 
+    /**
+     * Takes ownership of ptr, adding it to the back of the vector.
+     *
+     * @param ptr Pointer to take ownership of.
+     * @param status Is set to U_MEMORY_ALLOCATION_ERROR if allocation of the
+     * needed pool capacity fails.
+     */
+    void adoptBack(T *ptr, UErrorCode &status) {
+        this->adopt(ptr, status);
+    }
+
     int32_t length() const {
         return this->fCount;
     }
 
-    T** getAlias() {
+    T** getAlias() const {
```

Comment:
Why are you making this method `const`?

Reply:
I want to call it from a method that is marked const (accessing a member instance of the MaybeStackVector class).

getAlias is const in the sense that it doesn't modify the fPool pointer, but the contents of the pool may of course be modified: I now recognise, from what you wrote, that that might be rather confusing, and "const" methods on container types should perhaps be "deeply const". I've now created an overloaded getAlias(), how does this look?